src/models/tag.ts

```diff
@@ -0,0 +1,360 @@
+   * @param id the ID to decode
+   * @returns the tag name text
+   */
+  public static decodeId(id: string): string {
+    const decoded = Buffer.from(id, 'base64').toString();
+    const replace = `${config.data.tagIdSuffix}$`;
+    const regex = new RegExp(replace);
+    return decoded.replace(regex, '');
+  }
+
+  /**
+   * Encode an ID from a tag's name text
+   * @param name the tag name text
+   * @returns the encoded ID
+   */
+  public static encodeId(name: string): string {
+    return Buffer.from(name + config.data.tagIdSuffix).toString('base64');
+  }
+
+  /**
+   * Associate one or more tags to a save
+   */
+  public async createTagSaveConnections(
+    inputs: SavedItemTagsInput[]
+  ): Promise<SavedItem[]> {
+    const creates: TagCreateInput[] = sanitizeTagCreateInput(
+      inputs.flatMap((input) =>
+        input.tags.map((name) => ({
+          savedItemId: input.savedItemId,
+          name,
+        }))
+      )
+    );
+    await this.tagService.insertTags(creates);
+    const saveIds = creates.map((_) => _.savedItemId);
+    return this.saveService.batchGetSavedItemsByGivenIds(saveIds);
+  }
+
+  /**
+   * Replace the tags associated with a save
+   */
+  public updateTagSaveConnections(
+    updates: SavedItemTagUpdateInput
+  ): Promise<SavedItem> {
+    const creates: TagCreateInput[] = updates.tagIds.map((tagId) => {
+      return {
+        name: TagModel.decodeId(tagId),
+        savedItemId: updates.savedItemId,
+      };
+    });
+    const sanitized = sanitizeTagCreateInput(creates);
+    const sanitizedIds = sanitized.map(({ savedItemId }) => savedItemId);
+    // Validate just in case
+    const deleteFromSaveId = new Set(sanitizedIds);
+    if (deleteFromSaveId.size != 1) {
+      throw new UserInputError('Cannot update Tags on multiple Saves');
+    }
+    return this.tagService.updateSavedItemTags(sanitized);
+  }
+
+  /**
+   * Replace the tags associated with one or more saves in
+   * in a single batch.
+   */
+  public replaceTagSaveConnections(tags: TagCreateInput[]) {
+    const sanitizedInput = sanitizeTagCreateInput(tags);
+    return this.tagService.replaceSavedItemTags(sanitizedInput);
+  }
+
+  /**
+   * Fetch a Tag by its ID
+   * @param id the ID of the Tag to retrieve
+   * @throws NotFoundError if the record does not exist
+   * @returns the Tag entity
+   */
+  public getById(id: string): Promise<Tag> {
+    const name = TagModel.decodeId(id);
+    const tag = this.tagService.getTagByName(name);
+    if (tag == null) {
+      throw new NotFoundError(`Tag with ID=${id} does not exist.`);
+    }
+    return tag;
+  }
+
+  /**
+   * Fetch multiple Tags by ID
+   * Will not throw a NotFoundError if a tag ID is missing,
+   * so the size of the response might differ from the number
+   * of IDs requested.
+   * @param ids array of tag IDs to retrieve
+   * @returns An array of Tag entities
+   */
+  public getManyById(ids: string[]): Promise<Tag[]> {
+    // TODO: Not found partial repsonse?
```

Comment:
i couldn't find the usage of this function..but assuming this is used in bulk calls, if one of them is invalid, we decided to fail the whole call..so we don't need to handle partial response

Reply:
I forgot to add this to the resolvers it looks like. thank you!